synapse/module_api/__init__.py

```diff
@@ -405,39 +397,47 @@ async def send_local_online_presence_to(self, users: Iterable[str]) -> None:
         Updates to remote users will be sent immediately, whereas local users will receive
         them on their next sync attempt.
 
-        Note that this method can only be run on the main or federation_sender worker
-        processes.
+        Note that this method can only be run on the main or worker processes that have the
+        ability to write to the presence stream.
         """
-        if not self._hs.should_send_federation():
+        if (
+            self._hs.config.worker_app
+            and self._hs._instance_name not in self._hs.config.worker.writers.presence
+        ):
             raise Exception(
                 "send_local_online_presence_to can only be run "
-                "on processes that send federation",
+                "on processes that have the ability to write to the"
+                "presence stream (this includes the main process)",
```

Comment:
This only works on the process configured to write presence, which by default is master. If its not master then this won't work on master

Reply:
Right, that is an important clarification to mention. Is it correct that multiple processes can be configured to write to the presence stream similar to event persisters?

Reply:
There is a limit of 1, at least for the moment. The only reason its a list is for consistency with everything else